src/js/utils/date-utils.js

```diff
@@ -169,103 +185,88 @@ var DATE_FORMATS = {
-         *
-         *~~~ javascript
-         * var context = {
-         *    datetime: '2016-10-14 08:00:00+00:00',
-         *    timezone: 'Pacific/Kiritimati',
-         *    language: 'ar',
-         *    format: 'DEFAULT_LONG_DATE_FORMAT'
-         *    };
-         * return_string = DateUtils.localize(context);
-         *~~~
-         *
-         * - datetime: {string} The (UTC) datetime server-side string.
-         * - modifier: {string} (optional) 'due', 'deadline:', etc..
-         * - timezone: {string} (optional) the user set time zone preference, or 'None' if unset.
-         * This should be in a format like 'Region/Locale', e.g. 'Atlantic/Azores'. Defaults to 'UTC'.
-         * - language: {string} (optional), the user set language, an ISO 639-1 language string, defaults to 'en'
-         * - format: {string} (optional), a format constant. Defaults to 'DEFAULT_DATE_TIME_FORMAT'.
+         * @param {string} format (optional) A format constant.
+         * @returns {string} A format constant.
          */
-        localize = function(context) {
-            // made a little longer than necessary for clarity
-            var localTime;
-            var displayTime = '';
-            var displayString;
-
-            if (isValid(context.datetime)) {
-                localTime = localizeTime(
-                    DateUtils.dateObject(context.datetime),
-                    userTimezone(context.timezone)
-                );
-                displayTime = timeDisplay(localTime, context.language, context.format);
-            } else {
-                displayTime = '';
+        _getDisplayFormat = function(format) {
+            var displayFormat = DATE_FORMATS.DEFAULT_DATE_TIME_FORMAT;
+            if (_isValid(format) && format in DATE_FORMATS) {
+                displayFormat = DATE_FORMATS[format];
             }
+            return displayFormat;
+        };
 
-            if (isValid(context.modifier)) {
-                displayString = StringUtils.interpolate(
-                    '{modifier} {date}',
-                    {date: displayTime, modifier: context.modifier}
-                );
-            } else {
-                displayString = displayTime;
-            }
-            return displayString;
+        /**
+         * Helper function to determine variable/context validity.
+         *
+         * @param {string} candidateVariable to test
+         * @returns {boolean}
+         */
+        _isValid = function(candidateVariable) {
+            return candidateVariable !== undefined
+                && candidateVariable !== ''
+                && candidateVariable !== 'Invalid date'
+                && candidateVariable !== 'None';
         };
 
         DateUtils = {
-            isValid: isValid,
-            dateObject: dateObject,
-            userTimezone: userTimezone,
+            localize: localize,
+            stringtoObject: stringtoObject,
+            _getSysTimezone: _getSysTimezone,
```

Comment:
The intention of this object is to return the public methods. I'd remove all the functions that are currently named starting with underscore.